File: aat-lib/src/main/java/ch/bailu/aat_lib/service/tracker/TrackCrashRestorer.java
package ch.bailu.aat_lib.service.tracker;

import java.io.IOException;

import ch.bailu.aat_lib.gpx.GpxList;
import ch.bailu.aat_lib.gpx.attributes.AutoPause;
import ch.bailu.aat_lib.preferences.system.SolidDataDirectory;
import ch.bailu.aat_lib.resources.Res;
import ch.bailu.aat_lib.xml.writer.GpxListWriter;
import ch.bailu.aat_lib.xml.parser.gpx.GpxListReader;
import ch.bailu.aat_lib.logger.AppLog;
import ch.bailu.aat_lib.util.fs.AppDirectory;
import ch.bailu.foc.Foc;

public final class TrackCrashRestorer{

    public TrackCrashRestorer (SolidDataDirectory sdirectory, int presetIndex) throws IOException {

        Foc source = AppDirectory.getLogFile(sdirectory);

        if (source.exists()) {

            GpxList track = readFile(source);
            if (track.getPointList().size() > TrackLogger.MIN_TRACKPOINTS) {
                // TODO message to gui
                AppLog.i(this, Res.str().tracker_restore());

                retstoreFile(sdirectory,track, presetIndex);

            }
            source.rm();
        }

    }


    private void retstoreFile(SolidDataDirectory sdirectory, GpxList track, int presetIndex) throws IOException {
        Foc target = TrackLogger.generateTargetFile(sdirectory, presetIndex);

        GpxListWriter writer=new GpxListWriter(track,target);
        writer.close();

    }







    private GpxList readFile(Foc remainingLogFile) {
        GpxListReader reader = new GpxListReader(remainingLogFile, AutoPause.NULL);
        return reader.getGpxList();
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/service/tracker/TrackCrashRestorer.kt
package ch.bailu.aat_lib.service.tracker

import ch.bailu.aat_lib.file.xml.parser.gpx.GpxListReaderXml
import ch.bailu.aat_lib.file.xml.writer.GpxListWriter
import ch.bailu.aat_lib.gpx.GpxList
import ch.bailu.aat_lib.gpx.attributes.AutoPause
import ch.bailu.aat_lib.logger.AppLog
import ch.bailu.aat_lib.preferences.system.SolidDataDirectory
import ch.bailu.aat_lib.resources.Res.str
import ch.bailu.aat_lib.service.tracker.TrackLogger.Companion.generateTargetFile
import ch.bailu.aat_lib.util.fs.AppDirectory.getLogFile
import ch.bailu.foc.Foc
import java.io.IOException

class TrackCrashRestorer(sdirectory: SolidDataDirectory, presetIndex: Int) {
    init {
        val source = getLogFile(sdirectory)
        if (source.exists()) {
            val track = readFile(source)
            if (track.pointList.size() > TrackLogger.MIN_TRACK_POINTS) {
                AppLog.i(this, str().tracker_restore())
                restoreFile(sdirectory, track, presetIndex)
            }
            source.rm()
        }
    }

    @Throws(IOException::class)
    private fun restoreFile(sdirectory: SolidDataDirectory, track: GpxList, presetIndex: Int) {
        val target = generateTargetFile(sdirectory, presetIndex)
        val writer = GpxListWriter(track, target)
        writer.close()
    }

    private fun readFile(remainingLogFile: Foc): GpxList {
        val reader = GpxListReaderXml(remainingLogFile, AutoPause.NULL)
        return reader.gpxList
    }
}
